mmv1/third_party/terraform/website/docs/r/compute_firewall_policy.html.markdown

```diff
@@ -0,0 +1,110 @@
+#
+#     ***     AUTO GENERATED CODE    ***    Type: DCL     ***
+#
+# ----------------------------------------------------------------------------
+#
+#     This file is managed by Magic Modules (https:#github.com/GoogleCloudPlatform/magic-modules)
+#     and is based on the DCL (https:#github.com/GoogleCloudPlatform/declarative-resource-client-library).
+#     Changes will need to be made to the DCL or Magic Modules instead of here.
+#
+#     We are not currently able to accept contributions to this file. If changes
+#     are required, please file an issue at https:#github.com/hashicorp/terraform-provider-google/issues/new/choose
+#
+# ----------------------------------------------------------------------------
+subcategory: "Compute"
+layout: "google"
+page_title: "Google: google_compute_firewall_policy"
+sidebar_current: "docs-google-compute-firewall-policy"
+description: |-
+  Creates a hierarchical firewall policy
+---
+
+# google\_compute\_firewall\_policy
+
+Hierarchical firewall policy rules let you create and enforce a consistent firewall policy across your organization. Rules can explicitly allow or deny connections or delegate evaluation to lower level policies. Policies can be created within organizations or folders.
+
+This resource should be generally be used with `google_compute_firewall_policy_association` and `google_compute_firewall_policy_rule`
+
+For more information see the [official documentation](https://cloud.google.com/vpc/docs/firewall-policies)
+
+## Example Usage
+
+```hcl
+resource "google_compute_firewall_policy" "default" {
+  parent      = "organizations/12345"
+  short_name  = "my-policy"
+  description = "Example Resource"
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `parent` -
+  (Required)
+  The parent of the firewall policy.
+  
+* `short_name` -
+  (Required)
+  User-provided name of the Organization firewall policy. The name should be unique in the organization in which the firewall policy is created. The name must be 1-63 characters long, and comply with RFC1035. Specifically, the name must be 1-63 characters long and match the regular expression [a-z]([-a-z0-9]*[a-z0-9])? which means the first character must be a lowercase letter, and all following characters must be a dash, lowercase letter, or digit, except the last character, which cannot be a dash.
+  
+
+
+- - -
+
+* `description` -
+  (Optional)
+  An optional description of this resource. Provide this property when you create the resource.
+  
+
+
+## Attributes Reference
+
+In addition to the arguments listed above, the following computed attributes are exported:
+
+* `id` - an identifier for the resource with format `locations/global/firewallPolicies/{{name}}`
+
+* `creation_timestamp` -
+  Creation timestamp in RFC3339 text format.
+  
+* `fingerprint` -
+  Fingerprint of the resource. This field is used internally during updates of this resource.
+  
+* `id` -
```

Comment:
We can't support fields with the name `id` because the builtin `id` takes precedence. I think we want a rule to rename it based on the resource kind, `policy_id` or `firewall_policy_id` here.